### PR TITLE
add `-p`/`--slippi-port` cmd line option for spectate server port

### LINKED ISSUE #419: When using multiple playback dolphins, every SlippiSpectateServer after the first fails to start
this is because there is no mechanism to try a different port then the default (there also doesn't appear to be any way to change the default):
https://github.com/project-slippi/Ishiiruka/blob/18ee3450ee1cfde20ada2547c4792dc863dcd075/Source/Core/Core/Slippi/SlippiSpectate.cpp#L281-L302

adding an Ishii command-line option to set the SlippiSpectateServer port would alleviate this problem

### PR DESCRIPTION
**Pull request by jmlee337**
fixes #419

this will allow developers to connect to the spectate server on multiple dolphin instances on the same machine

### CODE REVIEW

**Review by NikhilNarayana — APPROVED** [review]

File: Source/Core/DolphinWX/Main.cpp [hunk 1]
```diff
@@ -141,6 +141,8 @@ bool DolphinApp::OnInit()
 
 	if (m_select_video_backend && !m_video_backend_name.empty())
 		SConfig::GetInstance().m_strVideoBackend = WxStrToStr(m_video_backend_name);
+	if (m_select_slippi_port && m_slippi_port > 0 && m_slippi_port < 65536)
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.cpp]:
```suggestion
	if (m_select_slippi_port && m_slippi_port > 1024 && m_slippi_port < 65536)
```

i just remembered that on linux you can't bind to a port under 1025 unless you're root. don't have a strong opinion on if we should ifdef it or apply it everywhere

**Review by NikhilNarayana — CHANGES_REQUESTED** [review]

File: Source/Core/DolphinWX/Main.cpp (hunk 1 above)
Comment (jmlee337) [reply Source/Core/DolphinWX/Main.cpp]: I'm seeing <1024, is that ok?

Comment (NikhilNarayana) [reply Source/Core/DolphinWX/Main.cpp]: yeah you're right, i misread it as 1-1024

**Review by NikhilNarayana — APPROVED** [review]

File: Source/Core/DolphinWX/Main.cpp [hunk 2]
```diff
@@ -320,6 +322,8 @@ void DolphinApp::OnInitCmdLine(wxCmdLineParser &parser)
 	     wxCMD_LINE_VAL_STRING, wxCMD_LINE_PARAM_OPTIONAL},
 	    {wxCMD_LINE_OPTION, "a", "audio_emulation", "Low level (LLE) or high level (HLE) audio", wxCMD_LINE_VAL_STRING,
 	     wxCMD_LINE_PARAM_OPTIONAL},
+	    {wxCMD_LINE_OPTION, "p", "slippi-port", "Port to use for the Slippi spectate server (1024 - 65535, default: 51441)",
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.cpp]:
```suggestion
	    {wxCMD_LINE_OPTION, nullptr, "slippi-spectator-port", "Port to use for the Slippi spectate server (1024 - 65535, default: 51441)",
```

File: Source/Core/DolphinWX/Main.h [hunk 3]
```diff
@@ -49,6 +49,7 @@ class DolphinApp : public wxApp
 	bool m_show_version = false;
 	bool m_select_video_backend = false;
 	bool m_select_slippi_input = false;
+	bool m_select_slippi_port = false;
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.h]:
```suggestion
	bool m_select_slippi_spectator_port = false;
```

File: Source/Core/DolphinWX/Main.h [hunk 4]
```diff
@@ -59,6 +60,7 @@ class DolphinApp : public wxApp
 	wxString m_video_backend_name;
 	wxString m_audio_emulation_name;
 	wxString m_slippi_input_name;
+	long m_slippi_port;
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.h]:
```suggestion
	long m_slippi_spectator_port;
```

File: Source/Core/DolphinWX/Main.cpp [hunk 5]
```diff
@@ -141,6 +141,8 @@ bool DolphinApp::OnInit()
 
 	if (m_select_video_backend && !m_video_backend_name.empty())
 		SConfig::GetInstance().m_strVideoBackend = WxStrToStr(m_video_backend_name);
+	if (m_select_slippi_port && m_slippi_port >= 1024 && m_slippi_port < 65536)
+		SConfig::GetInstance().m_spectator_local_port = m_slippi_port;
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.cpp]:
```suggestion
	if (m_select_slippi_spectator_port && m_slippi_spectator_port >= 1024 && m_slippi_spectator_port < 65536)
		SConfig::GetInstance().m_spectator_local_port = m_slippi_spectator_port;
```

File: Source/Core/DolphinWX/Main.cpp [hunk 6]
```diff
@@ -389,6 +393,7 @@ bool DolphinApp::OnCmdLineParsed(wxCmdLineParser &parser)
 	m_confirm_stop = parser.Found("confirm", &m_confirm_setting);
 	m_select_video_backend = parser.Found("video_backend", &m_video_backend_name);
 	m_select_audio_emulation = parser.Found("audio_emulation", &m_audio_emulation_name);
+	m_select_slippi_port = parser.Found("slippi-port", &m_slippi_port);
```
Comment (NikhilNarayana) [inline Source/Core/DolphinWX/Main.cpp]:
```suggestion
	m_select_slippi_spectator_port = parser.Found("slippi-spectator-port", &m_slippi_spectator_port);
```

**Review by NikhilNarayana — CHANGES_REQUESTED** [review]
apologies for suddenly changing my mind, i re-read it and decided i'd rather have this be very explicit about what port it refers to

**Comment by jmlee337** [comment]
looks like there's a precedent for 'slippi spectate' as in `SlippiSpectateServer`: https://github.com/project-slippi/Ishiiruka/blob/a9c5519a7bf7122e1d11fde7499538af988cac1c/Source/Core/Core/Slippi/SlippiSpectate.h#L39 is this ok?

**Comment by NikhilNarayana** [comment]
> looks like there's a precedent for 'slippi spectate' as in `SlippiSpectateServer`:
> 
> https://github.com/project-slippi/Ishiiruka/blob/a9c5519a7bf7122e1d11fde7499538af988cac1c/Source/Core/Core/Slippi/SlippiSpectate.h#L39
> is this ok?

not necessarily against it, but i would like the corresponding config value to be renamed to match so it is easy to match them up, if you prefer just `spectate`

**Comment by jmlee337** [comment]
oh the config name lol oops. No I'll go spectator I don't want to change the config name that could be breaking if people are touching it programmatically

**Review by NikhilNarayana — APPROVED** [review]

**Pull request merged**